src/main/java/com/aws/greengrass/tes/TokenExchangeService.java

```diff
@@ -94,6 +95,8 @@ protected void startup() {
             logger.atInfo().log("Started server at port {}", server.getServerPort());
             // Get port from the server, in case no port was specified and server started on a random port
             setEnvVariablesForDependencies(server.getServerPort());
+            // Store the actual port being used in the config so that the CLI can show the value for debugging
```

Comment:
Huh so the actual port could be different than the configured PORT_TOPIC?

Reply:
port topic contains the user's requested port (0 by default). Since 0 is the default, the port topic doesn't contain the real port since the OS will choose a random open port.